h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -270,8 +276,10 @@ public static ParseSetup guessSetup( Key[] fkeys, ParseSetup userSetup ) {
     if (ice instanceof Frame && ((Frame) ice).vec(0) instanceof UploadFileVec) {
       t._gblSetup._chunk_size = FileVec.DFLT_CHUNK_SIZE;
     } else {
-      t._gblSetup._chunk_size = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
-              Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+      int calculatedChunkSize = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
+          Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+      int finalChunkSize = Math.max(t._gblSetup.dataOffset + 2, calculatedChunkSize);
```

Comment:
@vpatryshev I think we are close to a merge. I still don't understand how the dataOffset works for multiple files, there is no handling of dataOffset in the setup unification method (the one that creates the global offset).

Also I think it needs to be very clearly explained why is this being done here, otherwise this knowledge will get lost over time.

 Would it really not be possible to change ARFFParser/CsvParser to support multichunk headers? How hard would that be?